src/compiler/parser.ts

```diff
@@ -6116,6 +6116,19 @@ namespace ts {
                     || node.kind === SyntaxKind.ExportDeclaration
                     ? node
                     : undefined);
+            if (isInJavaScriptFile(sourceFile) && !sourceFile.externalModuleIndicator) {
+                sourceFile.commonJsModuleIndicator = forEach(sourceFile.statements, isCommonJsModuleIndicator);
+            }
+        }
+
+        function isCommonJsModuleIndicator(statement: Statement) {
+            if (isExpressionStatement(statement) && isBinaryExpression(statement.expression)) {

```

Comment:
this is concerning.. first it is work that were not doing before, and second, it is potentially incomplete. in the binder we bind all require calls, not just the top-level ones, and not just binary expressions. so to do this right we need to walk the whole file until we find an indicator.. but we are doing this in the binder already.. 

is there a different way we can collect these typedefs then bind their export symbols at the end?

Reply:
That sounds like a good approach -- less lossy although less obvious. I'll see if I can get something working.